game-core/src/test/java/games/strategy/triplea/ai/pro/util/MoveBatcherTest.java

```diff
@@ -0,0 +1,130 @@
+package games.strategy.triplea.ai.pro.util;
+
+import static games.strategy.triplea.delegate.GameDataTestUtil.americans;
+import static games.strategy.triplea.delegate.GameDataTestUtil.armour;
+import static games.strategy.triplea.delegate.GameDataTestUtil.infantry;
+import static games.strategy.triplea.delegate.GameDataTestUtil.territory;
+import static games.strategy.triplea.delegate.GameDataTestUtil.transport;
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+
+import com.google.common.collect.Maps;
+import games.strategy.engine.data.GameData;
+import games.strategy.engine.data.Route;
+import games.strategy.engine.data.Territory;
+import games.strategy.engine.data.Unit;
+import games.strategy.triplea.xml.TestMapGameData;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.List;
+import java.util.Map;
+import org.junit.jupiter.api.Test;
+
+public class MoveBatcherTest {
+  private final GameData gameData = TestMapGameData.REVISED.getGameData();
+  private final Territory brazil = territory("Brazil", gameData);
+  private final Territory sz19 = territory("19 Sea Zone", gameData);
+  private final Territory sz18 = territory("18 Sea Zone", gameData);
+  private final Territory sz12 = territory("12 Sea Zone", gameData);
+  private final Territory algeria = territory("Algeria", gameData);
+  private final Route brazilToSz18 = new Route(brazil, sz18);
+  private final Route sz19ToSz18 = new Route(sz19, sz18);
+  private final Route sz18ToSz12 = new Route(sz18, sz12);
+  private final Route sz12ToAlgeria = new Route(sz12, algeria);
+  private final Route algeriaToSz18 = new Route(sz12, algeria);
+  private final Unit inf1 = infantry(gameData).create(americans(gameData));
+  private final Unit inf2 = infantry(gameData).create(americans(gameData));
+  private final Unit tank1 = armour(gameData).create(americans(gameData));
+  private final Unit tank2 = armour(gameData).create(americans(gameData));
+  private final Unit transport1 = transport(gameData).create(americans(gameData));
+  private final Unit transport2 = transport(gameData).create(americans(gameData));
+
+  public MoveBatcherTest() throws Exception {}
```

Comment:
This constructor is redundant

Reply:
It's needed because some of the initializing of the variables above can throw Exception, so compiler requires that the constructor is declared to throw it.

Reply:
Moving gameData to this constructor would also require moving everything else - since they reference gameData.

This would double the number of lines required, so I prefer not to do that. Let me know if you disagree.

Reply:
Ah I see, thanks for the clarification.
No, keeping everything as-is seems cleaner to me.
Not sure if it's worth adding a comment explaining the java mechanism, doesn't matter too much